### PR TITLE
Introduce common connection dialog page

### PR DESCRIPTION
**Pull request by rosswbrown**
- V1 and V2 details on their own pages
- Update auth-modes to be common guidance for both V1 and V2
- Move V1 specific content from auth-modes to ui.md
- Update left side navigation bar

### CODE REVIEW

File: docs/ui.md [hunk 1]
```diff
@@ -2,30 +2,32 @@
 title: Build the Connection Dialog with Connection Dialog v1
```
Comment (pvanderknyff) [inline docs/ui.md:2]: Probably a good idea to change the title of this page to match what it is in the index, same with the Connection Dialog v2 page

File: docs/connection-dialog.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+These elements are defined in the manifest.xml file:
+```xml
+<connector-plugin class='postgres_odbc' superclass='odbc' plugin-version='0.0.0' name='PostgreSQL ODBC' version='20.1'>
+  <vendor-information>
+      <company name="Company Name"/>
+      <support-link url = "http://example.com"/>
+  </vendor-information>
+  ...
+</connector-plugin>
```
Comment (pvanderknyff) [inline docs/connection-dialog.md:27]: Missing the driver download link, I'm pretty sure that is shown when trying to launch a connector where the driver is not installed nowadays.

File: docs/ui.md [hunk 3]
```diff
@@ -2,30 +2,32 @@
...
 ---
 
-The connection dialog prompts the user to enter connection and authentication information. That information is passed into the Connector Builder script to build the connection string. The dialog appears when creating a new connection or editing an existing connection and is used by both Tableau Desktop and Tableau Server.
-
-The connection dialog is mainly defined in the Tableau Custom Dialog (.tcd) file.
+## Define Tableau Custom Dialog file elements
```
Comment (pvanderknyff) [inline docs/ui.md:5]: Do we want to have a note on this page that we recommend V2, or do we think that the message in the new connection-dialog page is sufficient?

File: docs/auth-modes.md [hunk 4]
```diff
@@ -12,196 +12,26 @@ A combination of Connection Dialog and Connection Resolver changes implement eac
...
 
-### No Authentication
-User is never prompted for credentials.
+| Name | value | Description |
+| - | - | - |
+| No Authentication | `auth-none` | User is never prompted for credentials |
+| Username only | `auth-user` | User is prompted for username during initial connection creation. |
+| Username and Password | `auth-user-pass` | User is prompted for username and password during initial connection creation, and password only when reconnecting to the data source |
+| Password only | `auth-pass` | User is prompted for password during initial connection creation and reconnecting to the data source |
+| OAuth | `oauth` | User is prompted with the default brower for [OAuth]({{ site.baseurl }}/docs/oauth) credentials during initial connection creation and reconnecting to the data source |
```
Comment (pvanderknyff) [inline docs/auth-modes.md:21]: We should document `auth-integrated` here as well. 

Also, should we integrate the Kerberos documentation found in [this sample](https://github.com/tableau/connector-plugin-sdk/tree/master/samples/scenarios/jdbc_kerberos) to this doc page? That's so buried I didn't know it existed until Devansh pointed it out to me.

**Review by pvanderknyff — APPROVED** [review]
LGTM

File: docs/connection-dialog.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+The connector is displayed as "[Display Name] by [Company Name]" in the connection dialog and connection list.
+
+"For support, contact [Company Name]" is displayed at the bottom left of the connector. Clicking this link sends the user to the support link defined in the manifest. This link also displays in error messages. The support link must use HTTPS to be packaged into a TACO file.
+
+These elements are defined in the manifest.xml file:
+```xml
+<connector-plugin class='postgres_odbc' superclass='odbc' plugin-version='0.0.0' name='PostgreSQL ODBC' version='20.1'>
+  <vendor-information>
+      <company name="Company Name"/>
+      <support-link url = "http://example.com"/>
```
Comment (sxlin) [inline docs/connection-dialog.md:24]: Change support link to https to match document above.

**Review by sxlin — APPROVED** [review]

**Pull request merged**